backup: support object lock in s3 (#10460)
pingcap/docs: br/backup-storage-S3.md

The Backup & Restore (BR) tool supports using Amazon S3 or other Amazon S3-compatible file storages as the external storage for backing up and restoring data.

> **Note:**
>

> > To back up data to an S3 bucket with Object Lock enabled, you need to use TiDB v6.3.0 or later.

Reviewer: ```suggestion
> To back up data to an S3 bucket with Object Lock enabled, make sure that your TiDB cluster is v6.3.0 or later.
```

> > To back up data to an S3 bucket with object locking enabled, you need to use TiDB v6.3.0 or later.

Reviewer 2: `object lock` seems to be proprietary. https://docs.aws.amazon.com/AmazonS3/latest/userguide/object-lock.html

Author: Fixed in 1bd7552

> > To back up data to an S3 bucket with object locking enabled, you need to use TiDB v6.3.0 or later.

Author: ```suggestion
> To back up data to an S3 bucket with Object Lock enabled, you need to use TiDB v6.3.0 or later.
```
